fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -849,15 +869,67 @@ module.exports = function(config, DB) {
                 return devices[0]
               })
               .then(function (device) {
+                t.deepEqual(device.sessionTokenId, newSessionTokenId, 'sessionTokenId updated')
                 t.equal(device.name, 'updated name', 'name updated')
-                t.equal(device.type, newDevice.type, 'type unchanged')
-                return db.deleteDevice(device.uid, device.id)
+                t.equal(device.type, deviceInfo.type, 'type unchanged')
+                t.equal(device.callbackURL, '', 'callbackURL updated')
+                t.deepEqual(device.callbackPublicKey, zeroBuffer32, 'callbackPublicKey updated')
+                return db.upsertDevice(ACCOUNT.uid, newUuid(), {
+                  sessionTokenId: newSessionTokenId,
+                  name: 'test device',
+                  createdAt: Date.now(),
+                  type: 'mobile',
+                  callbackURL: 'https://foo/bar',
+                  callbackPublicKey: hex32()
+                })
+                .then(function() {
+                  t.fail('adding a second device does not fail when the session token is already registered')
```

Comment:
A query about wording here.  Every time I read these, I parse them as statements of desired behaviour.  So I just read the above, thought "adding a second device does not fail when already registered? that's doesn't sound right..." and took a minute to realize that it was *failing* if that was the case.  What do you think about writing these failure guards in past tense, so it's more obvious that they're telling you that something went wrong? 

e.g. "adding a second device should have failed when..."